src/pages/Swap/index.tsx

```diff
@@ -198,21 +198,25 @@ export default function Swap() {
     if (!swapCallback) {
       return
     }
-    setSwapState({ attemptingTxn: true, tradeToConfirm, showConfirm, swapErrorMessage: undefined, txHash: undefined })
+    setSwapState(prevState => ({ ...prevState, attemptingTxn: true, swapErrorMessage: undefined, txHash: undefined }))
     swapCallback()
       .then(hash => {
-        setSwapState({ attemptingTxn: false, tradeToConfirm, showConfirm, swapErrorMessage: undefined, txHash: hash })
+        setSwapState(prevState => ({
+          ...prevState,
+          attemptingTxn: false,
+          swapErrorMessage: undefined,
+          txHash: hash
+        }))
```

Comment:
In these cases I removed the dependencies because they were just updating themselves (from the same state tree). My guess is the person who did this wanted to avoid the Typescript error (or didn't understand how dependencies worked).